src/inference/tempering.jl

```diff
@@ -0,0 +1,166 @@
+        sampler.N_swap,
+        sampler.swap_strategy
+    )
+end
+
+
+function AbstractMCMC.sample(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    sampler::MCMCTempering.TemperedSampler{<:InferenceAlgorithm},
+    N::Integer;
+    chain_type=MCMCChains.Chains,
+    resume_from=nothing,
+    progress=PROGRESS[],
+    kwargs...
+)
+    tempered_sampler = create_tempered_sampler(model, sampler)
+    if resume_from === nothing
+        return AbstractMCMC.sample(rng, model, tempered_sampler, N;
+                                   chain_type=chain_type, progress=progress, kwargs...)
+    else
+        return resume(resume_from, N; chain_type=chain_type, progress=progress, kwargs...)
+    end
+end
+
+
+function AbstractMCMC.sample(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    sampler::MCMCTempering.TemperedSampler{<:InferenceAlgorithm},
+    ensemble::AbstractMCMC.AbstractMCMCEnsemble,
+    N::Integer,
+    n_chains::Integer;
+    chain_type=MCMCChains.Chains,
+    progress=PROGRESS[],
+    kwargs...
+)
+    tempered_sampler = create_tempered_sampler(model, sampler)
+    return AbstractMCMC.sample(rng, model, tempered_sampler, ensemble, N, n_chains;
+                               chain_type=chain_type, progress=progress, kwargs...)
+end
+
+
+
+###########################
+#   MODEL
+###########################
+
+struct TemperedEval{T<:Real}
+    model :: DynamicPPL.Model
+    β     :: T
+end
+
+function (te::TemperedEval)(
+    rng::Random.AbstractRNG,
+    model::DynamicPPL.Model,
+    varinfo::DynamicPPL.AbstractVarInfo,
+    sampler::DynamicPPL.AbstractSampler,
+    context::DynamicPPL.AbstractContext,
+    args...
+)
+    minibatchcontext = DynamicPPL.MiniBatchContext(
+        context,
+        te.β
+    )
+    te.model.f(rng, model, varinfo, sampler, minibatchcontext, args...)
+end
+
+"""
+    MCMCTempering.make_tempered_model
+
+Uses a `TemperedEval` to temper a `model`'s `f` by using a `DynamicPPL.MiniBatchContext` to inject `β` as a multiplier on evaluation
+"""
+function MCMCTempering.make_tempered_model(model::DynamicPPL.Model, β::Real)
+    return DynamicPPL.Model(model.name, TemperedEval(model, β), model.args, model.defaults)
+end
+
+
+
+###########################
+#   SWAPPING
+###########################
+
+function MCMCTempering.get_tempered_loglikelihoods_and_params(
+    model::DynamicPPL.Model,
+    sampler::DynamicPPL.Sampler{<:InferenceAlgorithm},
+    states,
+    k::Integer,
+    Δ::Vector{T},
+    Δ_state::Vector{<:Integer}
+) where {T<:AbstractFloat}
+
+    logπk = MCMCTempering.make_tempered_loglikelihood(model, Δ[Δ_state[k]], sampler, get_vi(states[k][2]))
+    logπkp1 = MCMCTempering.make_tempered_loglikelihood(model, Δ[Δ_state[k + 1]], sampler, get_vi(states[k + 1][2]))
+    
+    θk = MCMCTempering.get_params(states[k][2], sampler)
+    θkp1 = MCMCTempering.get_params(states[k + 1][2], sampler)
+    
+    return logπk, logπkp1, θk, θkp1
+end
+
+
+"""
+    MCMCTempering.make_tempered_loglikelihood
```

Comment:
According to the convention and Julia style guide this should be the function signature (and usually lists of arguments etc. should be avoided).